connectors/kafka-connector/src/main/scala/zio/connect/kafka/KafkaConnector.scala

```diff
@@ -0,0 +1,30 @@
+package zio.connect.kafka
+
+import org.apache.kafka.clients.producer.{ProducerRecord}
+import org.apache.kafka.common.Uuid
+import zio.Trace
+import zio.connect.kafka.KafkaConnector._
+import zio.kafka.consumer._
+import zio.stream._
+
+trait KafkaConnector {
+  def createTopic(implicit trace: Trace): ZSink[Any, Throwable, NewTopic, Nothing, Unit]
+
+  def listTopics(implicit trace: Trace): ZStream[Any, Throwable, TopicListing]
+
+  def read(topic: => String)(implicit trace: Trace): ZStream[Any, Throwable, CommittableRecord[String, String]]
+  def write(implicit trace: Trace): ZSink[Any, Throwable, ProducerRecord[String, String], ProducerRecord[String, String], Unit]

```

Comment:
I think it's better not use ProducerRecord type here. It ties the api to zio-kafka. A custom type that in the live implementation is converted to a ProducerRecord would be preferable.

Reply:
Same comment as for the CommittableRecord one but this one actually a kafka client class and not a zio-kafka class.